enclave/core/sgx/threadlocal.c

```diff
@@ -332,6 +332,24 @@ oe_result_t oe_thread_local_init(oe_sgx_td_t* td)
             _thread_locals_relocated = true;
         }
 
+        {
+            // Currently this is the most appropriate spot to initialize the
```

Comment:
This doesn't explain why it's a problem for `oe_allocator_init()` to be called during `_handle_init_enclave()` after `td_init` though. Consider:

```
// oe_allocator_init needs to be invoked once globally before oe_allocator_thread_init
// can be called. Since oe_allocator_thread_init needs to be invoked on each td_init,
// we call oe_allocator_init here with a run-once lock instead of during _handle_init_enclave,
// which is tool late since that only gets to run after td_init.
```
Also, why not just simplify this by calling it through `oe_once()`?

Reply:
Updated comment. Nice suggestion about using oe_once. Done.